### ISSUE TITLE
Avoid clade collapsing corner-case with large-angle clades in the circular / unrooted layouts

### ISSUE THREAD

**Issue by fedarko** [biocore/empress#351]
Copied from #277:

> It should also be noted that an issue can occur with the Unrooted/Circular layout if the clade is more than a 180°:
> ![circ-bug](https://user-images.githubusercontent.com/28127694/88436689-a212b400-cdb9-11ea-85cd-2b6cd8b9eb23.png)
> ![unroot-lay](https://user-images.githubusercontent.com/28127694/88436695-a4750e00-cdb9-11ea-843f-a0f32cdc948c.png)
>
> This is not a bug per se but a consequence of the collapsing algorithm (I believe iTol shares the same issue). This happens because the "left" and "right" most children of the clade are behind the root. This, in turn, causes the clade shape to do one of two things: point in the wrong direction (this happened in the above Unrooted layout) or overlap other clades/root (this happened in the circular layout).

It should be pretty clear when this bug is occurring, but it'd be nice to anticipate it and handle it gracefully (to avoid people getting confused and including the pac-man-looking shapes in papers). A few tentative options:

- **For the circular layout** (maybe this could work for the unrooted layout also): Detect if the total angle span in the collapsed clade is > 180° (or just > pi if we're in radians) and, if so, just draw a small conservative shape that doesn't overlap any other nodes (maybe just using the symmetric collapsing code, maybe another way if needed). Maybe we could also pop up a toast warning message so the user knows the width of the collapsed clade will be arbitrary for the circular layout.
- Rather than drawing wedges for the circular layout, approximate the wedges with a bunch of quadrilaterals?

TLDR, it's a pretty minor bug, but it'd be nice to make things more foolproof if we can IMO.
